streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -488,6 +488,7 @@ private WindowStoreReadWriteDecorator(final WindowStore<K, V> inner) {
         }
 
         @Override
+        @SuppressWarnings("deprecation")
```

Comment:
Recommend propagating the deprecation, rather than suppressing it.
```suggestion
        @Deprecated
```